src/hooks/useProfiles.ts

```diff
@@ -3,24 +3,31 @@ import { UseMutationOptions, UseQueryOptions, useMutation, useQuery } from 'reac
 import { useNavigate } from 'react-router';
 import { useTranslation } from 'react-i18next';
 
-import { createProfile, deleteProfile, enterProfile, listProfiles, updateProfile } from '#src/stores/ProfileController';
 import { useProfileStore } from '#src/stores/ProfileStore';
 import type { CommonAccountResponse, ListProfilesResponse, ProfileDetailsPayload, ProfilePayload } from '#types/account';
 import { useAccountStore } from '#src/stores/AccountStore';
 import type { GenericFormErrors } from '#types/form';
 import type { ProfileFormSubmitError, ProfileFormValues } from '#src/containers/Profiles/types';
 import { logDev } from '#src/utils/common';
+import ProfileController from '#src/stores/ProfileController';
+import { getModule } from '#src/modules/container';
+import AccountController from '#src/stores/AccountController';
 
 export const useSelectProfile = () => {
   const navigate = useNavigate();
+  const isLoggedIn = useAccountStore((s) => !!s.user);
 
-  return useMutation((vars: { id: string; pin?: number; avatarUrl: string }) => enterProfile({ id: vars.id, pin: vars.pin }), {
+  const accountController = isLoggedIn ? getModule(AccountController) : undefined;
+  const profileController = isLoggedIn ? getModule(ProfileController) : undefined;
+
+  return useMutation(async (vars: { id: string; pin?: number; avatarUrl: string }) => profileController?.enterProfile({ id: vars.id, pin: vars.pin }), {
     onMutate: ({ avatarUrl }) => {
       useProfileStore.setState({ selectingProfileAvatar: avatarUrl });
     },
     onSuccess: async () => {
       useProfileStore.setState({ selectingProfileAvatar: null });
       navigate('/');
+      await accountController?.loadUserData();
```

Comment:
Where is all of this extra logic in this file coming from? Is it a merge artifact?

Reply:
It is not. ProfileController used `loadUserData` for `enterProfile` method which created unnecessary connection with AccountController (AccountController already has a profile dependecy).

Reply:
Oh OK! Good refactor!